ingest/indexer/service/blockprocessor/block_updates_indexer_block_process_strategy.go

```diff
@@ -24,29 +24,34 @@ func (f *blockUpdatesIndexerBlockProcessStrategy) IsFullBlockProcessor() bool {
 // ProcessBlock implements commondomain.BlockProcessStrategy.
 func (f *blockUpdatesIndexerBlockProcessStrategy) ProcessBlock(ctx types.Context) error {
 	// Publish supplies
-	if err := f.publishChangedPools(ctx); err != nil {
+	if err := f.publishCreatedPools(ctx); err != nil {
 		return err
 	}
 
 	return nil
 }
 
 // publishChangedPools publishes the pools that were changed in the block.
-func (f *blockUpdatesIndexerBlockProcessStrategy) publishChangedPools(ctx types.Context) error {
+func (f *blockUpdatesIndexerBlockProcessStrategy) publishCreatedPools(ctx types.Context) error {
 	err := f.blockUpdateProcessUtils.ProcessBlockChangeSet()
 	if err != nil {
 		return err
 	}
 	// Extract the pools that were changed in the block
-	blockPools, err := f.poolExtractor.ExtractChanged(ctx)
+	blockPools, createdPoolIDs, err := f.poolExtractor.ExtractCreated(ctx)
 	if err != nil {
 		return err
 	}
 
 	pools := blockPools.GetAll()
 
+	// Do nothing if no pools were created, or pool metadata is nil
+	if len(createdPoolIDs) == 0 || len(pools) == 0 {
+		return nil
+	}
+
 	// Publish pool pairs
-	if err := f.poolPairPublisher.PublishPoolPairs(ctx, pools); err != nil {
+	if err := f.poolPairPublisher.PublishPoolPairs(ctx, pools, createdPoolIDs); err != nil {
```

Comment:
Can we apply the creation timestamps and meta data from `createdPoolIDs` before propagating it to `poolPairPublisher.PublishPoolPairs`?

The following are the reasons for the request:
1. `poolPairPublisher.PublishPoolPairs` should be concerned about 1 thing only - that is, publishing pool pairs. Currently, we do some data transformations, going aginst the separation of concerns design principle while polluting the API with another parameter that is not universally needed.
2. The method is already quite complex and untested

Instead, we can apply the "creation metadata transformations" in the caller. 

Also, would appreciate creating a task for testing `poolPairPublisher.PublishPoolPairs`